plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -333,6 +333,38 @@ public Transaction newCreateTableTransaction(
                 Optional.of(session.getUser()));
     }
 
+    @Override
+    public Transaction newReplaceTableTransaction(
+            ConnectorSession session,
+            SchemaTableName schemaTableName,
+            Schema schema,
+            PartitionSpec partitionSpec,
+            String location,
+            Map<String, String> properties)
+    {
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
+                schemaTableName,
+                ignore -> {
+                    TableOperations operations = tableOperationsProvider.createTableOperations(
+                            this,
+                            session,
+                            schemaTableName.getSchemaName(),
+                            schemaTableName.getTableName(),
+                            Optional.empty(),
+                            Optional.empty());
+                    return new BaseTable(operations, quotedTableName(schemaTableName)).operations().current();
+                });
```

Comment:
Is this copied from `loadTable`?
Do those lines need to stay in sync in the future?